api/controller/assignments.js

```diff
@@ -0,0 +1,321 @@
+                    res.send({ 'error': err });
+                });
+
+            }
+
+        });
+    });
+};
+
+
+function createEntry(entryDoc) {
+    var objectId = common.createUUID();
+    if (typeof entryDoc == "object") {
+        entryDoc.objectId = objectId;
+    }
+    return entryDoc;
+}
+
+function updateEntry(entryDoc, privateJournalId) {
+    return new Promise(function (resolve, reject) {
+        db.collection(journalCollection, function (err, collection) {
+            collection.updateOne(
+                {
+                    _id: new mongo.ObjectID(privateJournalId)
+                },
+                {
+                    $push: {
+                        "content": entryDoc
+                    }
+
+                },
+                function (err, result) {
+                    if (err) {
+                        reject(err);
+                    } else {
+                        resolve(result);
+                    }
+                }
+            );
+
+        });
+    });
+}
+
+exports.removeAssignment = function (req, res) {
+    //validate
+    if (!req.params.assingid) {
+        res.status(400).send({
+            error: "Assignment id is not defined",
+            code: 22
+        });
+        return;
+    }
+
+    db.collection(assignmentCollection, function (err, collection) {
+        collection.deleteOne(
+            {
+                _id: new mongo.ObjectID(req.params.assingid)
+            },
+            function (err, result) {
+                if (err) {
+                    res.status(500).send({
+                        error: "An error has occurred",
+                        code: 10
+                    });
+                } else {
+                    if (result && result.result && result.result.n == 1) {
+                        res.send({
+                            id: req.params.assingid
+                        });
+                    } else {
+                        res.status(401).send({
+                            error: "Inexisting assignment id",
+                            code: 23
+                        });
+                    }
+                }
+            }
+        );
+    });
+};
+
+exports.updateAssignment = function (req, res) {
+    //validate
+    if (!req.params.assingid) {
+        res.status(400).send({
+            error: "Assignment id is not defined",
+            code: 22
+        });
+        return;
+    }
+    var assingid = req.params.assingid;
+    console.log(req.body.assignment);
+    var assignment = JSON.parse(req.body.assignment);
+
+    //add timestamp
+    assignment.timestamp = +new Date();
+
+    //find assignment by id
+    db.collection(assignmentCollection, function (err, collection) {
+        //count data
+        collection.countDocuments(
+            {
+                '_id': {
+                    $ne: new mongo.ObjectID(assingid)
+                },
+                'name': new RegExp("^" + assignment.name + "$", "i")
```

Comment:
Two assignments can have same names. You don't need this filter logic. Simply filter by matching the assignmentId.